File: video/ai-extensions/video-80-leetcode-5/leetcode5.swift

class Solution {
    func convert(_ s: String, _ numRows: Int) -> String {
        guard numRows > 1 else { return s }
        
        var rows = Array(repeating: "", count: min(numRows, s.count))
        var currentRow = 0
        var goingDown = false
        
        for char in s {
            rows[currentRow] += String(char)
            if currentRow == 0 || currentRow == numRows - 1 {
                goingDown.toggle()
            }
            currentRow += goingDown ? 1 : -1
        }
        
        return rows.joined()
    }
}

File: video/ai-extensions/video-80-leetcode-5/leetcode5.ts

function convert(s: string, numRows: number): string {
  if (numRows === 1) return s;

  const rows: string[] = new Array(numRows).fill("");
  let currentRow = 0;
  let goingDown = false;

  for (let char of s) {
      rows[currentRow] += char;
      if (currentRow === 0 || currentRow === numRows - 1) goingDown = !goingDown;
      currentRow += goingDown ? 1 : -1;
  }

  return rows.join("");
}
